uvicorn/supervisors/multiprocess.py

```diff
@@ -90,9 +90,13 @@ def kill(self) -> None:
         # In Unix, the method will send SIGKILL to the process.
         self.process.kill()
 
-    def join(self) -> None:
+    def join(self, join_timeout: float | None = None) -> None:
         logger.info(f"Waiting for child process [{self.process.pid}]")
-        self.process.join()
+        self.process.join(join_timeout)
+        # Timeout, kill the process
+        while self.process.exitcode is None:
+            self.process.kill()
+            self.process.join(1)
```

Comment:
Why do we have a `join(1)` here?

Reply:
Wait for the kill command to take effect. If it does not take effect within 1 second, send the kill command again.